Fix snippet substitution issue

FireChickenProductivity opened the pull request.

dsheyp commented:
I briefly tested this PR and found that it fixed [cursorless issue #3075](https://github.com/cursorless-dev/cursorless/issues/3075)

FireChickenProductivity commented:
Testing Summary:
I have tested my changes to:
sql.py
rust.py
r.py
lua.py
css.py

Stuff I have tested with snippets (using Java):
Wrapping: 
"try wrap line"
"if wrap line" (tests $0 with wrapping)
"if cond wrap line" (tests $1 with wrapping)

Modified snippet insertion:
"snip if"
"snip else"
"snip return"
"snip ternary" (corner case: $0 already at the end)
"snip class"
"snip class test name"
"snip break" (corner case: no snippet number present)
"snip ternary" with the body I modified to "$1 ? $2 : $3" (corner case: no $0 with snippet numbers present)

Original snipped insertion actions:
"snip if"
"snip else"
"snip return"
"snip ternary"
"snip class"
"snip class test name"
"snip break"

nriley reviewed (changes requested):
If you could just add a quick comment in BREAKING_CHANGES to mention the change in default behavior and how you can return to it either for a single snippet or for all snippets as desired comment that would be great. Thanks.
